tests/codegen/llvm/array_integer_equal_comparison.ll

```diff
@@ -63,80 +64,71 @@ if_body:                                          ; preds = %arraycmp.done
 if_end:                                           ; preds = %deadcode, %arraycmp.done
   ret i64 0
 
-arraycmp.false:                                   ; preds = %arraycmp.loop7, %arraycmp.loop3, %arraycmp.loop, %entry
+while_cond:                                       ; preds = %arraycmp.loop, %entry
```

Comment:
I don't think that this is correct. Looking at [array_integer_equal_comparison.cpp](https://github.com/iovisor/bpftrace/blob/master/tests/codegen/array_integer_equal_comparison.cpp#L10-L13):
```cpp
  if (bpftrace->has_loop())
  {
    return;
  }
```
This test should be skipped when the machine has the `bpf_loop` helper, so, IIUC, the codegen should have the loop unrolled, which is what the original version has.

FWIW, I think that we have a bug somewhere in the testing logic b/c I've seen this on my machine, too, with the old `update_codegen_tests.sh`.

Reply:
Good catch. I concur, I think there is a bug in that conditional test logic. I've put up #2714 which should improve the situation.